### PR TITLE
Modify rule S6229: Improve description

### PR DESCRIPTION
**Pull request by marco-antognini-sonarsource**
Fix a few typos and add the "noncompliant solution" section.

### CODE REVIEW

File: rules/S6229/cfamily/rule.adoc [hunk 1]
```diff
@@ -6,11 +6,51 @@ The ``++chrono++`` library, introduced in {cpp}20, provides support for calendar
...
-* querying for current time (``++time++``, ``++get_timespec++``, ``++clock_gettime++``)
+* querying for current time (``++time++``, ``++timespec_get++``, ``++clock_gettime++``)
 * date to time-point conversion (``++mktime++``, ``++gmtime++``, ``++localtime++``)
 * time serialization (``++ctime++``, ``++asctime++``, ``++strftime++``)
 * time parsing (``++strptime++``)
 
+== Noncompliant Solution
+
+----
+int currentMonth() {
```
Comment (loic-joly-sonarsource) [inline rules/S6229/cfamily/rule.adoc:17]: I think we usually use 2 spaces for indentation in our code examples (even if we're not always consistent...).

File: rules/S6229/cfamily/rule.adoc [hunk 2]
```diff
@@ -6,11 +6,51 @@ The ``++chrono++`` library, introduced in {cpp}20, provides support for calendar
...
+    std::time_t tp;
+    std::time(&tp);
+    std::tm* date = std::gmtime(&tp);
+    return date->tm_mon + 1;
+}
+
+std::chrono::system_clock::time_point makeSomeDay() {
+   // Creates time_point corresponding to 2001-04-10
```
Comment (loic-joly-sonarsource) [inline rules/S6229/cfamily/rule.adoc:25]: Maybe a more recent date would be appropriate... Unless it's your date of birth?
I suggest 4 September 2020, (date of approbation of C++20)

Comment (marco-antognini-sonarsource) [reply rules/S6229/cfamily/rule.adoc:25]: Haha no, I'm not that young 😅 . I like your date tough, so I'll update the snippets to match.

File: rules/S6229/cfamily/rule.adoc (hunk 1 above)
Comment (marco-antognini-sonarsource) [reply rules/S6229/cfamily/rule.adoc:17]: The description should now be more consistent.

**Review by loic-joly-sonarsource — APPROVED** [review]

File: rules/S6229/cfamily/rule.adoc [hunk 3]
```diff
@@ -6,36 +6,76 @@ The ``++chrono++`` library, introduced in {cpp}20, provides support for calendar
...
+  using namespace std::chrono;
+  auto dp = floor<days>(system_clock::now());
+  return year_month_day(dp).month();
 }
 
 std::chrono::system_clock::time_point makeSomeDay() {
-   using namespace std::chrono;
-   retrun sys_days(2001y/April/10);
+  using namespace std::chrono;
+  return sys_days(2020y/Septembre/4);
```
Comment (arseniy-sonar) [inline rules/S6229/cfamily/rule.adoc:66]: I believe you meant `September`
`std::chrono` has no `Septembre`

Comment (marco-antognini-sonarsource) [reply rules/S6229/cfamily/rule.adoc:66]: Good catch! Will open a PR to fix this right away.

**Pull request merged**